server/mediaserver.go

```diff
@@ -154,19 +162,59 @@ func createRTMPStreamIDHandler(s *LivepeerServer) func(url *url.URL) (strmID str
 			mid = core.RandomManifestID()
 		}
 
+		var auth bool
+		if mid, auth = authenticateStream(url.String(), string(mid)); !auth {
+			glog.Error("Authentication denied for ", url)
+			return ""
+		}
+
 		// Ensure there's no concurrent StreamID with the same name
 		s.connectionLock.RLock()
-		defer s.connectionLock.RUnlock()
```

Comment:
We can probably leave the `defer` here since subsequent operations are not expensive. Less error prone

Reply:
done